crates/turbopack-tests/tests/execution.rs

```diff
@@ -0,0 +1,345 @@
+        .cell();
+
+    let context: AssetContextVc = ModuleAssetContextVc::new(
+        TransitionsByNameVc::cell(HashMap::new()),
+        compile_time_info,
+        ModuleOptionsContext {
+            enable_jsx: Some(JsxTransformOptionsVc::cell(JsxTransformOptions {
+                development: true,
+                ..Default::default()
+            })),
+            enable_typescript_transform: Some(TypescriptTransformOptionsVc::default()),
+            preset_env_versions: Some(env),
+            rules: vec![(
+                ContextCondition::InDirectory("node_modules".to_string()),
+                ModuleOptionsContext {
+                    ..Default::default()
+                }
+                .cell(),
+            )],
+            custom_ecma_transform_plugins: Some(CustomEcmascriptTransformPluginsVc::cell(
+                CustomEcmascriptTransformPlugins {
+                    source_transforms: vec![TransformPluginVc::cell(
+                        Box::<StyledJsxTransformer>::default(),
+                    )],
+                    output_transforms: vec![],
+                },
+            )),
+            ..Default::default()
+        }
+        .into(),
+        ResolveOptionsContext {
+            enable_typescript: true,
+            enable_react: true,
+            enable_node_modules: Some(project_root),
+            custom_conditions: vec!["development".to_string()],
+            rules: vec![(
+                ContextCondition::InDirectory("node_modules".to_string()),
+                ResolveOptionsContext {
+                    enable_node_modules: Some(project_root),
+                    custom_conditions: vec!["development".to_string()],
+                    ..Default::default()
+                }
+                .cell(),
+            )],
+            ..Default::default()
+        }
+        .cell(),
+    )
+    .into();
+
+    let chunk_root_path = path.join("output");
+    let static_root_path = path.join("static");
+    let chunking_context = DevChunkingContextVc::builder(
+        project_root,
+        chunk_root_path,
+        chunk_root_path,
+        static_root_path,
+        env,
+    )
+    .build();
+
+    let modules: Vec<EvaluatableAssetVc> = entry_paths
+        .into_iter()
+        .map(SourceAssetVc::new)
+        .map(|p| {
+            EvaluatableAssetVc::from_asset(
+                context.process(
+                    p.into(),
+                    Value::new(ReferenceType::Entry(EntryReferenceSubType::Undefined)),
+                ),
+                context,
+            )
+        })
+        .collect();
+
+    let entry = context.process(
+        SourceAssetVc::new(jest_entry_asset).into(),
+        Value::new(ReferenceType::Entry(EntryReferenceSubType::Undefined)),
+    );
+
+    let res = evaluate(
+        entry,
+        chunk_root_path,
+        CommandLineProcessEnvVc::new().into(),
+        modules.first().unwrap().ident(),
```

Comment:
Nit: can't we get this from the `entry`? Or the `jest_runtime_asset` directly (which is what `modules.first().unwrap()` gives us)